### PR TITLE
sqlite: update to v3.41.0

### PR DESCRIPTION
**Pull request by akabanovs**
-sha256 updated.
 -dependency pinnings updated.

### CODE REVIEW

**Comment by akabanovs** [comment]
There are ncurses 6.3 and 6.4 available on all archs. However, it looks like 6.3 has a CVE, which is why I decided to pin ncurses to 6.4. Please let me know if this is the right decision.

**Review by SC426 — COMMENTED** [review]
I would remove the `license_url` since you have `license_file` specified.

You can also remove `doc_source_url` since you have `doc_url` specified.

**Review by varlackc — COMMENTED** [review]
As soon as the previous comments are addressed we should be good to approve the pull request.

**Comment by akabanovs** [comment]
3.41.1 came out with some bug fixes. 
https://www.sqlite.org/releaselog/3_41_1.html

The recipe has been updated.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,12 +33,12 @@ requirements:
...
-    - readline  # [not win]
-    - zlib      # [not win]
+    - ncurses   6.4  # [not win]
+    - readline  {{readline}} # [not win]
+    - zlib      {{zlib}}     # [not win]
   run:
-    - readline  # [not win]
-    - zlib      # [not win]
+    - {{ pin_compatible('readline') }}  # [not win]
+    - {{ pin_compatible('zlib') }}      # [not win]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:41]: @katietz I wonder why we didn't use `pin_compatible` earlier? Are there any drawbacks?

Comment (katietz) [reply recipe/meta.yaml:41]: well, pin compatible has the disadvantage of relying on the pretty random run_export of the used readline package.  So having this stuff more explicit in recipe is benefitial, as more easy to understand and reliable

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by katietz — APPROVED** [review]
I removed the ignore_run_exports for ncurse here.  We should keep the run-dependency
I changed logic to use explicit cbc pin values for packages, and use pin_compatible for run (as we don't know what minimum version is picked up).
lgtm otherwise

**Pull request merged**